### ISSUE TITLE
Errors after upgrade from 4.12.5 to 4.13.0

### ISSUE THREAD

**Issue by jjyy1946** [marko-js/marko#1156]
### Details

After upgraded from 4.12.5 to 4.13.0 I got following errors with the same code:
Uncaught DOMException: Failed to execute 'insertBefore' on 'Node': Nodes of type '#text' may not be inserted inside nodes of type '#document'.
    at insertBefore (http://localhost:9000/static/js/bundle.js:80147:23)
    at createFragmentNode (http://localhost:9000/static/js/bundle.js:80112:5)
    at indexServerComponentBoundaries (http://localhost:9000/static/js/bundle.js:78975:36)
    at HTMLDocument.<anonymous> (http://localhost:9000/static/js/bundle.js:79229:21)
insertBefore @ helpers.js:5
createFragmentNode @ fragment.js:62
indexServerComponentBoundaries @ init-components-browser.js:44
(anonymous) @ init-components-browser.js:298

Tried to debug, but so far no success on identify the root cause. Can someone give a point on how to debug something like this?
Thanks
Yi
### Possible Fix

<details><summary>Additional Info</summary>

### Your Environment

- Environment name and version (e.g. Chrome 39, node.js 5.4):
- Operating System and version (desktop or mobile):
- Link to your project:

### Steps to Reproduce

1.  first...
2.
3.
4.

### Stack Trace

</details>

**Comment by nthibert** [comment]
@jjyy1946 I was wondering if you by chance found a solution to this issue. I have just upgraded and got the same issue.

**Comment by jasonmacdonald** [comment]
Any chance someone can write a failing test for this? They really help to expedite finding a solution to the issue.

**Comment by nthibert** [comment]
I have been trying to dig into this issue a bit more. From what I can see is that it's failing in the helper.js. 
`function insertBefore(node, referenceNode, parentNode) {
    if (node.insertInto) {
        return node.insertInto(parentNode, referenceNode);
    }
    return parentNode.insertBefore(node, referenceNode && referenceNode.startNode || referenceNode);
}`

This is exception is happening on the second return. What the values are:
parentNode: #document
node: ""
referenceNode: <!doctype html>
referenceNode.startNode: undefined

So I believe the issue comes when trying to the init the app on the whole html document and that text node can not be inserted into the document.

**Comment by ccinelli** [comment]
This is a bad error that has been improved in : #1266
> Stateful components cannot contain `<html>`, `<head>` or `<body>` tags. More details: https://github.com/marko-js/marko/wiki/Error:-Stateful-components-cannot-contain--html-,--head--or--body--tag

The work around is to refactor your code so you move out the stateful `app`. Ex: 

```
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8"/>
        <title>
            <${data.title}/>
        </title>
        <lasso-head/>
    </head>
    
    <body class="no-touch">

         <-- This is stateful -->
         <app />

        <lasso-body/>
        <browser-refresh/>
    </body>
</html>
```

**Comment by DylanPiercey** [comment]
This should be fixed in [4.18.16](https://github.com/marko-js/marko/blob/master/CHANGELOG.md#41816)
